src/consolidated_notation.rs

```diff
@@ -186,26 +186,28 @@ pub enum PrintingError {
     NumChildrenChanged,
     #[error("Pretty printing encountered a Text or Literal after an EndOfLine.")]
     TextAfterEndOfLine,
+    #[error("PrettyDoc implementation error: {0}")]
```

Comment:
I'd just call this `PrettyDoc error:`. I realize the word `implementation` is trying to distinguish the PrettyDoc trait from one particular implementation of it. But putting `implementation` in there makes it sound like the pretty printer has detected a mistake in the PrettyDoc trait (which is common, some of the other errors have that form).